pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/MethodReturnsInternalArray.xml

```diff
@@ -266,4 +266,116 @@ public enum MethodReturnsInternalArrayCaseEnum {
         ]]></code>
     </test-code>
 
+    <test-code>
+        <description>Inner interface</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public interface OuterInterface {
+       interface InnerInterface {
+            String method();
```

Comment:
The method should actually return a array, otherwise we are testing not our rule here :smile:

```java
public interface OuterInterface {
       interface InnerInterface {
            String[] method();
       }
}
```

Reply:
I should not commit anything after midnight :/
fixed